### PR TITLE
Unblock Accept after closing listener

### PR DESCRIPTION
**Pull request by mxpv**
Hello. This fixes issue with blocked Accept after closing listener. Details are here #19

### CODE REVIEW

**Comment by mdlayher** [comment]
Will review now, but /cc @aybabtme since he's running this software too.

File: vsock_test.go [hunk 1]
```diff
@@ -44,3 +48,45 @@ func TestAddr_fileName(t *testing.T) {
...
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		t.Log("start accept")
+		_, err := listener.Accept()
+		t.Log("after accept")
+		if err != nil {
+			return
```
Comment (mdlayher) [inline vsock_test.go]: Do something in this block?  Maybe panic if you get an error other than one caused by the socket closing?

File: vsock_test.go [hunk 2]
```diff
@@ -44,3 +48,45 @@ func TestAddr_fileName(t *testing.T) {
...
+		}
+	}()
+
+	time.Sleep(1 * time.Second)
```
Comment (mdlayher) [inline vsock_test.go]: I'm always a little hesitant to add a sleep in a test to make it pass, but I also don't have any better ideas at the moment.  Maybe @aybabtme will.

File: vsock_test.go (hunk 1 above)
Comment (mxpv) [reply vsock_test.go]: Done

**Comment by mxpv** [comment]
@mdlayher @aybabtme is there anything else I can do to get this merged?

File: vsock_test.go (hunk 2 above)
Comment (aybabtme) [reply vsock_test.go]: I'd make this be:

```go
select {
case <-time.After(time.Second):
    t.Fail("should have unblocked")
    return
case <-done:
    // continue
}
```
this way the test case only sleeps for 1s when there's a problem, as opposed to _always_ sleeping for 1s

Comment (aybabtme) [reply vsock_test.go]: 1s seems too high, maybe 100ms so it doesnt perceivably slow down the test. otherwise, idt there's a good alternative...

**Comment by aybabtme** [comment]
lgtm, sorry for not looking at this earlier. perhaps dial down the 1s sleep to 100ms

**Comment by mxpv** [comment]
@aybabtme done!

**Comment by mdlayher** [comment]
LGTM and apologies for the delay. Thanks!

**Pull request merged**